packages/framework-provider-aws/src/library/subscription-adapter.ts

```diff
@@ -71,40 +74,46 @@ export async function deleteSubscription(
-    .query({
-      TableName: config.resourceNames.subscriptionsStore,
-      IndexName: subscriptionsStoreAttributes.indexByConnectionIDName(config),
-      KeyConditionExpression:
-        `${subscriptionsStoreAttributes.indexByConnectionIDPartitionKey} = :partitionKey AND ` +
-        `${subscriptionsStoreAttributes.indexByConnectionIDSortKey} = :sortKey`,
-      ExpressionAttributeValues: {
-        ':partitionKey': connectionID,
-        ':sortKey': subscriptionID,
-      },
-    })
-    .promise()
+  let result: PromiseResult<DynamoDB.DocumentClient.QueryOutput, AWSError>
 
-  const foundSubscriptions = result.Items as Array<SubscriptionIndexRecord>
-  if (foundSubscriptions?.length < 1) {
-    logger.info(
-      `[deleteSubscription] No subscriptions found with connectionID=${connectionID} and subscriptionID=${subscriptionID}`
-    )
-    return
-  }
+  do {
+    result = await db
+      .query({
+        TableName: config.resourceNames.subscriptionsStore,
+        IndexName: subscriptionsStoreAttributes.indexByConnectionIDName(config),
+        KeyConditionExpression:
+          `${subscriptionsStoreAttributes.indexByConnectionIDPartitionKey} = :partitionKey AND ` +
+          `${subscriptionsStoreAttributes.indexByConnectionIDSortKey} = :sortKey`,
+        ExpressionAttributeValues: {
+          ':partitionKey': connectionID,
+          ':sortKey': subscriptionID,
+        },
+      })
+      .promise()
 
-  const subscriptionToDelete = foundSubscriptions[0] // There can't be more than one, as we used the full primary key in the query
-  logger.debug('[deleteSubscription] Deleting subscription = ', subscriptionToDelete)
-  await db
-    .delete({
-      TableName: config.resourceNames.subscriptionsStore,
-      Key: {
-        [subscriptionsStoreAttributes.partitionKey]: subscriptionToDelete[subscriptionsStoreAttributes.partitionKey],
-        [subscriptionsStoreAttributes.sortKey]: subscriptionToDelete[subscriptionsStoreAttributes.sortKey],
-      },
-    })
-    .promise()
+    const foundSubscriptions = result.Items as Array<SubscriptionIndexRecord>
+    if (foundSubscriptions?.length < 1) {
+      logger.info(
+        `[deleteSubscription] No subscriptions found with connectionID=${connectionID} and subscriptionID=${subscriptionID}, querying again`
+      )
+      break
+    }
+
+    const subscriptionToDelete = foundSubscriptions[0] // There can't be more than one, as we used the full primary key in the query
+    logger.debug('[deleteSubscription] Deleting subscription = ', subscriptionToDelete)
+    await db
+      .delete({
+        TableName: config.resourceNames.subscriptionsStore,
+        Key: {
+          [subscriptionsStoreAttributes.partitionKey]: subscriptionToDelete[subscriptionsStoreAttributes.partitionKey],
+          [subscriptionsStoreAttributes.sortKey]: subscriptionToDelete[subscriptionsStoreAttributes.sortKey],
+        },
+      })
+      .promise()
+  } while (result.LastEvaluatedKey)
```

Comment:
Why do we need this loop here? I think we are dealing with only one subscription and we are querying with the full primary key, so there won't be any need for pagination here (as far as I can tell)

Reply:
Because DynamoDB will run the query for the first 100 items, and will return something if its in that page, we have to keep querying until we run out of items to query. If not, it could be that we run the query, its not on the first 100 items, but maybe on the next, but we think its not subscribed.

Not sure if I explained this properly.